### PR TITLE
test: harden EventsBySliceFirehoseSpec, #31979

### PR DESCRIPTION
**Pull request by patriknw**
References https://github.com/akka/akka/issues/31979

### CODE REVIEW

File: akka-persistence-query/src/test/scala/akka/persistence/query/typed/internal/EventsBySliceFirehoseSpec.scala [hunk 1]
```diff
@@ -265,6 +268,7 @@ class EventsBySliceFirehoseSpec
       catchupPublisher(1).sendNext(allEnvelopes(0))
       outProbe(0).expectNext(allEnvelopes(0))
       outProbe(1).expectNext(allEnvelopes(0))
+
       firehosePublisher.sendNext(allEnvelopes(0))
       // this "sleep" is needed so that the firehose envelope is received first
       outProbe.expectNoMessage()
```
Comment (patriknw) [inline akka-persistence-query/src/test/scala/akka/persistence/query/typed/internal/EventsBySliceFirehoseSpec.scala:274]: seems like I had some similar problem with another test

File: akka-persistence-query/src/test/scala/akka/persistence/query/typed/internal/EventsBySliceFirehoseSpec.scala [hunk 2]
```diff
@@ -223,12 +223,15 @@ class EventsBySliceFirehoseSpec
...
 
-      firehosePublisher.sendNext(allEnvelopes(0))
       catchupPublisher(0).sendNext(allEnvelopes(0))
       catchupPublisher(1).sendNext(allEnvelopes(0))
       outProbe(0).expectNext(allEnvelopes(0))
       outProbe(1).expectNext(allEnvelopes(0))
 
+      firehosePublisher.sendNext(allEnvelopes(0))
+      // this "sleep" is needed so that the firehose envelope is received first
+      outProbe.expectNoMessage()
```
Comment (patriknw) [inline akka-persistence-query/src/test/scala/akka/persistence/query/typed/internal/EventsBySliceFirehoseSpec.scala:233]: not great that it's timing sensitive, but don't know an easy way to solve that

Comment (octonato) [reply akka-persistence-query/src/test/scala/akka/persistence/query/typed/internal/EventsBySliceFirehoseSpec.scala:233]: I don't think we can do better than that.

**Review by octonato — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Pull request merged**